### PR TITLE
[MultiTenant] MultiTenant framework

### PR DESCRIPTION
**Pull request by luchsh**
Summary: ported MultiTenant framework to Dragonwell8

Test Plan: jdk/test/multi-tenant

Reviewed-by: yuleil, sanhong

Issue: https://github.com/alibaba/dragonwell8/issues/84

### CODE REVIEW

**Comment by luchsh** [comment]
Please review https://github.com/alibaba/dragonwell8_hotspot/pull/29 firstly, this PR depends on that one

File: make/mapfiles/libjava/reorder-x86 [hunk 1]
```diff
@@ -10,6 +10,8 @@ text: .text%collapse: OUTPUTDIR/canonicalize_md.o;
 text: .text%Java_java_lang_Object_registerNatives;
 text: .text%Java_java_lang_System_registerNatives;
 text: .text%Java_java_lang_Thread_registerNatives;
+text: .text%Java_com_alibaba_tenant_TenantGlobals_getTenantFlags;
+text: .text%Java_com_alibaba_tenant_TenantGlobals_getTenantFlags;
```
Comment (superajun-wsj) [inline make/mapfiles/libjava/reorder-x86:14]: the two added lines seem the same, please delete one.

File: make/mapfiles/libjava/reorder-x86 [hunk 2]
```diff
@@ -107,4 +109,4 @@ text: .text%Java_java_lang_Class_isInstance;
 # SwingSet
 text: .text%Java_java_util_TimeZone_getSystemTimeZoneID;
 text: .text%findJavaTZ_md;
-text: .text%Java_java_lang_StrictMath_log;
+text: .text%Java_java_lang_StrictMath_log;
```
Comment (superajun-wsj) [inline make/mapfiles/libjava/reorder-x86:112]: it's an unnecessary change.

File: src/share/classes/com/alibaba/tenant/TenantConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ *
+ * The configuration used by tenant
+ */
+public class TenantConfiguration {
+
+    /*
+     * Resource throttling configurations
+     */
+    private Map<ResourceType, Constraint> constraints = new HashMap<>();
```
Comment (superajun-wsj) [inline src/share/classes/com/alibaba/tenant/TenantConfiguration.java:43]: is there concurrency security risk for constraints?

File: src/share/classes/com/alibaba/tenant/TenantContainer.java [hunk 4]
```diff
@@ -0,0 +1,660 @@
...
+    private void checkKey(String key) {
+        if (null == key) {
+            throw new NullPointerException("key can't be null");
+        }
+        if ("".equals(key)) {
+            throw new IllegalArgumentException("key can't be empty");
+        }
+    }
+
+    //
```
Comment (superajun-wsj) [inline src/share/classes/com/alibaba/tenant/TenantContainer.java:205]: should remove '//' here.

File: src/share/classes/com/alibaba/tenant/TenantContainer.java [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+    /**
+     * Gets the cpu time consumed by this tenant
+     * @return the cpu time used by this tenant, 0 if tenant cpu throttling or accounting feature is disabled.
+     */
+    public long getProcessCpuTime() {
+        if (!TenantGlobals.isCpuAccountingEnabled()) {
+            throw new IllegalStateException("-XX:+TenantCpuAccounting is not enabled");
+        }
+        long cpuTime = 0;
+        return cpuTime;
```
Comment (superajun-wsj) [inline src/share/classes/com/alibaba/tenant/TenantContainer.java:421]: it seems the return value is always 0.

File: src/share/classes/com/alibaba/tenant/TenantResourceType.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.alibaba.tenant;
+
+import com.alibaba.rcm.Constraint;
+import com.alibaba.rcm.ResourceType;
+import java.util.Arrays;
+import java.util.stream.Collectors;
```
Comment (superajun-wsj) [inline src/share/classes/com/alibaba/tenant/TenantResourceType.java:27]: should we remove unused import package?

File: make/mapfiles/libjava/reorder-x86 (hunk 1 above)
Comment (luchsh) [reply make/mapfiles/libjava/reorder-x86:14]: done

File: make/mapfiles/libjava/reorder-x86 (hunk 2 above)
Comment (luchsh) [reply make/mapfiles/libjava/reorder-x86:112]: removed

File: src/share/classes/com/alibaba/tenant/TenantConfiguration.java (hunk 3 above)
Comment (luchsh) [reply src/share/classes/com/alibaba/tenant/TenantConfiguration.java:43]: it is user's responsibility to ensure concurrency correctness

File: src/share/classes/com/alibaba/tenant/TenantResourceType.java (hunk 6 above)
Comment (luchsh) [reply src/share/classes/com/alibaba/tenant/TenantResourceType.java:27]: done
